cub/device/device_merge_sort.cuh

```diff
@@ -994,6 +994,152 @@ struct DeviceMergeSort
+   * @param[in] stream
+   *   **[optional]** CUDA stream to launch kernels within. Default is
+   *   stream<sub>0</sub>.
+   *
+   * [Random Access Iterator]: https://en.cppreference.com/w/cpp/iterator/random_access_iterator
+   * [Strict Weak Ordering]: https://en.cppreference.com/w/cpp/concepts/strict_weak_order
+   * [LessThan Comparable]: https://en.cppreference.com/w/cpp/named_req/LessThanComparable
+   */
+  template <typename KeyInputIteratorT,
+            typename KeyIteratorT,
+            typename OffsetT,
+            typename CompareOpT>
+  CUB_RUNTIME_FUNCTION static cudaError_t
+  StableSortKeysCopy(void *d_temp_storage,
+                     std::size_t &temp_storage_bytes,
+                     KeyInputIteratorT d_input_keys,
+                     KeyIteratorT d_output_keys,
+                     OffsetT num_items,
+                     CompareOpT compare_op,
+                     cudaStream_t stream = 0)
+  {
+    return SortKeysCopy<KeyInputIteratorT, KeyIteratorT, OffsetT, CompareOpT>(d_temp_storage,
+                                                                              temp_storage_bytes,
+                                                                              d_input_keys,
+                                                                              d_output_keys,
+                                                                              num_items,
+                                                                              compare_op,
+                                                                              stream);
+  }
+
+  template <typename KeyInputIteratorT,
+            typename KeyIteratorT,
+            typename OffsetT,
+            typename CompareOpT>
+  CUB_DETAIL_RUNTIME_DEBUG_SYNC_IS_NOT_SUPPORTED
+      CUB_RUNTIME_FUNCTION static cudaError_t
+      StableSortKeysCopy(void *d_temp_storage,
+                         std::size_t &temp_storage_bytes,
+                         KeyInputIteratorT d_input_keys,
+                         KeyIteratorT d_output_keys,
+                         OffsetT num_items,
+                         CompareOpT compare_op,
+                         cudaStream_t stream,
+                         bool debug_synchronous)
+  {
+    CUB_DETAIL_RUNTIME_DEBUG_SYNC_USAGE_LOG
+
+    return StableSortKeysCopy<KeyInputIteratorT, KeyIteratorT, OffsetT, CompareOpT>(d_temp_storage,
+                                                                                    temp_storage_bytes,
+                                                                                    d_input_keys,
+                                                                                    d_output_keys,
+                                                                                    num_items,
+                                                                                    compare_op,
+                                                                                    stream);
+  }
+
```

Comment:
```suggestion
```

This is not required. The `debug_synchronous` overloads exist so as not to break API. For the new functions, we don't have anything to break, so let's remove this overload.